src/services/resource-service.ts

```diff
@@ -43,22 +42,38 @@ export const ResourceService = {
       })
     })
   },
-  getLesson: async (id?: string): Promise<AxiosResponse<Lesson>> =>
-    await axiosClient.get(createUrlPath(URLs.resources.lessons.get, id)),
+  getLesson: async (id: string) => {
+    return baseService.request<Lesson>({
+      method: 'GET',
+      url: getFullUrl({
+        pathname: URLs.resources.lessons.getById,
+        parameters: { id }
+      })
+    })
+  },
   deleteLesson: async (id: string): Promise<AxiosResponse<Lesson>> =>
     await axiosClient.delete(createUrlPath(URLs.resources.lessons.delete, id)),
-  addLesson: async (data: LessonData): Promise<AxiosResponse> =>
-    await axiosClient.post(URLs.resources.lessons.add, data),
-  editLesson: async (data: LessonData, id?: string): Promise<AxiosResponse> =>
-    await axiosClient.patch(
-      createUrlPath(URLs.resources.lessons.patch, id),
+  addLesson: async (data: LessonData) => {
+    return baseService.request<Lesson>({
+      method: 'POST',
+      url: URLs.resources.lessons.add,
       data
-    ),
+    })
+  },
+  editLesson: async (data: LessonData, id: string) => {
+    return baseService.request<Lesson>({
```

Comment:
Check the backend response, probably there is nothing in body.
```suggestion
    return baseService.request<void>({
```